demos/TicTacToe/Classes/main_menu_scene.cc

```diff
@@ -51,6 +51,9 @@ bool MainMenuScene::init() {
   // Call the function to initialize the firebase features.
   this->InitializeFirebase();
 
+  // Initializes the event_dispatcher_ for this layer.
+  event_dispatcher_ = Director::getInstance()->getEventDispatcher();
```

Comment:
I'm not sure it's a good idea to have a long term pointer to the event dispatcher.  My previous comment was in the case where you were making to method invocations on the dispatcher in a row. There you can just create a variable on the stack (ie a temporary variable). I would roll back these private event_dispatcher_ changes, unforutnately, though it does make the code look better :) it **might** actually cause memory corruption issues :(